### ISSUE TITLE
[FEA]How to write a custom training loop from scratch in Tensorflow(v2.3).

### ISSUE THREAD

**Issue by sanchitvj** [gimseng/99-ML-Learning-Projects#127]
**Is your feature request related to a problem? Please describe.**
For the people who want to learn the Tensorflow framework and get the experience of how to use it from scratch.  

**Describe the solution you'd like**
I've recently done a project in which I entirely wrote code by myself. It will help people to deal with the situation when they've to write a training loop from scratch. I've done an image dehazing project in which the data set is images.  

**Describe alternatives you've considered**
So in that project, I've explained how to deal when dataset and labels both are images, in this case, hazy images are input and original images are labels. The code is in Tensorflow version 2.3.  

**Additional context**
Link to repo: https://github.com/sanchitvj/Image-Dehazing-using-GMAN-net

**Comment by gimseng** [comment]
Hi @sanchitvj thanks for the suggestion. I assume  you meant to propose a GMAN net exercise ?
I didn't quite get what you meant by "write a custom training loop".

**Comment by sanchitvj** [comment]
Yes, you got my point, let me clarify your doubt. I meant here is to explain how to write a training loop by ourselves in the situations when we can't use fit, predict with dataset in Tensorflow. Like sometimes in situations when our dataset is semi-supervised learning based so we have to write a training loop by ourselves. Example: [check here](https://github.com/sanchitvj/Image-Dehazing-using-GMAN-net/blob/master/train.py).

**Comment by gimseng** [comment]
@sanchitvj I see, I got what you meant. That sounds like a nice illustration of the training loop. 

Do read the guidelines for contributions and take a look at some of our previous projects so to understand how to organize and format your PR. Thanks !

**Comment by sanchitvj** [comment]
@gimseng I've sent the PR. Can you please check if it is suitably organized for merging?
